embulk-jruby-strptime/src/main/java/org/embulk/spi/time/StrptimeParser.java

```diff
@@ -0,0 +1,883 @@
+package org.embulk.spi.time;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.util.EnumSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.embulk.spi.time.lexer.StrptimeLexer;
+
+/**
+ * This is Java implementation of ext/date/date_strptime.c in Ruby v2.3.1.
+ * @see https://github.com/ruby/ruby/blob/394fa89c67722d35bdda89f10c7de5c304a5efb1/ext/date/date_strptime.c
+ *
+ * TODO
+ * This class is tentatively required for {@code TimestampParser} class.
+ * After embulk-jruby-strptime will be merged into JRuby, embulk-jruby-strptime
+ * will be removed from Embulk.
```

Comment:
To be honest, I'm not very confident if we should remove these even once these are merged into JRuby.

It is still an option to keep them as our own "Ruby-compatible date/time parser implemented in pure-Java" in Embulk so that the Embulk core does not have any dependency on JRuby. Using JRuby's implementation means that the Embulk core's behavior may be unintentionally affected by JRuby's future updates.

Contributing to JRuby is very nice, but it's worth being careful to depend on that in our very internal data structure.